lib/experimental/Forms/Fields/Select/index.tsx

```diff
@@ -1,31 +1,100 @@
+import * as Icons from "@/icons"
+import { cn, focusRing } from "@/lib/utils"
 import {
   SelectContent,
-  SelectItem,
+  SelectItem as SelectItemPrimitive,
   Select as SelectPrimitive,
   SelectTrigger,
-  SelectValue,
+  SelectValue as SelectValuePrimitive,
 } from "@/ui/select"
 import { forwardRef } from "react"
 
+type IconName = keyof typeof Icons
+
+type SelectItemProps<T> = {
+  value: T
+  label: string
+  icon?: IconName
+  description?: string
+}
+
 type SelectProps<T> = {
   placeholder: string
   onChange: (value: T) => void
   value?: T
-  options: { value: T; label: string }[]
+  options: SelectItemProps<T>[]
+  children?: React.ReactNode
+  disabled?: boolean
+}
+
+const SelectItem = ({ item }: { item: SelectItemProps<string> }) => {
+  const Icon = item.icon && Icons[item.icon]
+  return (
+    <SelectItemPrimitive value={item.value}>
+      <div className="flex items-start gap-1.5">
+        {Icon && <Icon className="h-5 w-5 text-f1-icon" />}
+        <div className="flex flex-col">
+          <span className="font-medium">{item.label}</span>
+          {item.description && (
+            <div className="text-f1-foreground-secondary">
+              {item.description}
+            </div>
+          )}
+        </div>
+      </div>
+    </SelectItemPrimitive>
+  )
 }
 
+const SelectValue = ({ item }: { item: SelectItemProps<string> }) => {
+  const Icon = item.icon && Icons[item.icon]
+  return (
+    <div className="flex items-center gap-1.5">
+      {Icon && <Icon className="h-5 w-5 text-f1-icon" />}
```

Comment:
should we put `shrink-0` here or we already have in on the Icon component level?

Reply:
Ah, good catch! The Icon component does have `shrink-0`, but as I'm using the icon directly it is not applied, so adding it.

Now that I think about it, I should be using the Icon component instead...